ext/symengine/ruby_complex.c

```diff
@@ -0,0 +1,23 @@
+#include "ruby_complex.h"
+
+VALUE ccomplex_init(VALUE self, VALUE comp_value) {
+    basic_struct *this;
+    basic real_basic, imag_basic;
+
+    basic_new_stack(real_basic);
+    basic_new_stack(imag_basic);
+
+    Data_Get_Struct(self, basic_struct, this);
+
+    VALUE real, imag;
+    real = rb_funcall(comp_value, rb_intern("real"), 0, NULL);
+    imag = rb_funcall(comp_value, rb_intern("imaginary"), 0, NULL);
+
+    crational_init(real_basic, real);
+    crational_init(imag_basic, imag);
```

Comment:
`crational_init` takes a `VALUE` and extracts the c pointer from that [in this line](https://github.com/symengine/symengine.rb/blob/master/ext/symengine/ruby_rational.c#L10). You are passing a `basic_struct *` which is being cast to [`VALUE`/`unintptr_t`/`void *`](http://blog.x-aeon.com/2012/12/13/the-ruby-c-api-basics/#TRCAB_Ruby).